python/ray/air/predictor.py

```diff
@@ -82,17 +80,13 @@ def predict(self, data: DataBatchType, **kwargs) -> DataBatchType:
         Returns:
             DataBatchType: Prediction result.
         """
+        data_df = convert_batch_type_to_pandas(data)
 
-        pass
+        if hasattr(self, "preprocessor") and self.preprocessor:
+            data_df = self.preprocessor.transform_batch(data_df)
 
-        # TODO(amogkam): Implement below code.
-        # data_df = _convert_batch_type_to_pandas(data)
-        #
-        # if hasattr(self, "preprocessor") and self.preprocessor:
-        #     data_df = self.preprocessor.transform_batch(data_df)
-        #
-        # predictions_df = self._predict_pandas(data_df)
-        # return _convert_pandas_to_batch_type(predictions_df, type=type(data))
+        predictions_df = self._predict_pandas(data_df, **kwargs)
+        return convert_pandas_to_batch_type(predictions_df, type=type(data))
 
     @DeveloperAPI
     def _predict_pandas(self, data: pd.DataFrame, **kwargs) -> pd.DataFrame:
```

Comment:
Can we have conditional imports here as well (in TYPE_CHECKING)? Otherwise the import in `air/__init__.py` will effectively introduce a dependency of Train and AIR on pandas.

Also happy with just accepting this dependency, but it seems this is currently the only place where this is leaked.

```suggestion
    def _predict_pandas(self, data: pd.DataFrame, **kwargs) -> "pd.DataFrame":
```

Reply:
Ah it seems with the data conversion util we can't really avoid this dependency (except we move this to a local import).

In that case we should probably update the Ray train dependencies and minimal install test.